### ISSUE TITLE
Why is an empty object literal (`{}`) considered immutable?

### ISSUE THREAD

**Issue by pmcelhaney** [danielnixon/eslint-plugin-total-functions#754]
This code fails `total-functions/no-unsafe-readonly-mutable-assignment` because `Unsafe "Immutable" to "Mutable" assignment. Source: "{}", destination: "SimpleRecord"`

```ts
const o: Record<string, string> = {};
```

But this code does not trigger the rule. 

```ts
const o: Record<string, string> = { x: "x" };
```

I'm confused. Why is an empty object literal considered immutable?

**Comment by danielnixon** [comment]
`{}` is immutable for the same reason that `{ readonly a: string }` is immutable - there is no way to mutate either value (at least not directly). It's just that one has one property and the other has zero properties.

Here's an example that shows the surprising mutation that can arise (the sort that this rule is designed to catch):

```ts
const foo = {};

console.log(foo);

// doesn't compile
// foo.a = "whoops";

type MutableThing = { a?: string };

// Assignment from immutable to mutable.
const bar: MutableThing = foo;

bar.a = "whoops";

// foo has mutated in a surprising way
console.log(foo);
```

What's special about your example is that the RHS is an object literal, not a value reference. In the special case where the RHS is an empty object literal (or empty array literal) we could update this rule not to complain. Assignment from an empty object literal to either a mutable or an immutable target is safe.

PR welcome.

**Comment by danielnixon** [comment]
Fixed in [v7.0.7](https://github.com/danielnixon/eslint-plugin-total-functions/commit/7aad7e661b7d1984a134e6f3638d1332791a0b83)
